src/services/statemanager.js

```diff
@@ -45,25 +45,60 @@ ngeo.StateManager = function(ngeoLocation) {
       for (i = 0; i < count; ++i) {
         key = this.localStorage.key(i);
         goog.asserts.assert(key !== null);
-        this.initialState[key] = this.localStorage.get(key);
+        this.initialState[key] = this.getItemFromLocalStorage_(key);
       }
+      this.ngeoLocation.updateParams(this.initialState);
     }
   } else {
     var keys = ngeoLocation.getParamKeys();
     for (i = 0; i < keys.length; ++i) {
       key = keys[i];
-      this.initialState[key] = ngeoLocation.getParam(key);
+      this.initialState[key] = this.getItemFromLocation_(key);
     }
   }
+};
+
 
-  this.ngeoLocation.updateParams({});
+/**
+ * Get the item for the given key  from localStorage and try to parse to the appropriate type
+ * If it cannot be parsed, the raw value (string) is returned
+ *
+ * @param  {string} key the localStorage key for the item
+ * @return {string|boolean|Object|number|undefined} Param value.
```

Comment:
Does it mean that it can be anything?
In this case, you can use `{*}` if I'm not wrong.